src/main/java/com/hms_networks/americas/sc/extensions/system/http/SCHttpUtility.java

```diff
@@ -284,6 +326,328 @@ public static String httpGet(String url, String header, String body, String outp
+   * @return http response string
+   * @since 1.15.2
+   */
+  public static String httpPatch(String url, String header, String body, String outputFile)
+      throws EWException,
+          IOException,
+          SCHttpEwonException,
+          SCHttpAuthException,
+          SCHttpConnectionException,
+          SCHttpUnknownException {
+    return httpRequest(url, header, body, HTTPX_PATCH_STRING, outputFile);
+  }
+
+  /**
+   * Performs an HTTP DELETE request to the specified URL using the specified request header and
+   * body.
+   *
+   * @param url URL to make request
+   * @param header request header
+   * @param body request body
+   * @throws EWException if unable to make request
+   * @throws IOException if unable to read response file
+   * @throws SCHttpEwonException if an Ewon error occurs during the request
+   * @throws SCHttpAuthException if an authentication error occurs during the request
+   * @throws SCHttpConnectionException if a connection error occurs during the request
+   * @throws SCHttpUnknownException if an unknown error occurs during the request
+   * @return http response string
+   * @since 1.15.2
+   */
+  public static String httpDelete(String url, String header, String body)
+      throws EWException,
+          IOException,
+          SCHttpEwonException,
+          SCHttpAuthException,
+          SCHttpConnectionException,
+          SCHttpUnknownException {
+    return httpRequest(url, header, body, HTTPX_DELETE_STRING, null);
+  }
+
+  /**
+   * Performs an HTTP DELETE request to the specified URL using the specified request header and
+   * body, then outputs the response to the specified file. If the specified file already exists, it
+   * will be overwritten.
+   *
+   * @param url URL to make request
+   * @param header request header
+   * @param body request body
+   * @param outputFile file to output response to
+   * @throws EWException if unable to make request
+   * @throws IOException if unable to read response file
+   * @throws SCHttpEwonException if an Ewon error occurs during the request
+   * @throws SCHttpAuthException if an authentication error occurs during the request
+   * @throws SCHttpConnectionException if a connection error occurs during the request
+   * @throws SCHttpUnknownException if an unknown error occurs during the request
+   * @return http response string
+   * @since 1.15.2
+   */
+  public static String httpDelete(String url, String header, String body, String outputFile)
+      throws EWException,
+          IOException,
+          SCHttpEwonException,
+          SCHttpAuthException,
+          SCHttpConnectionException,
+          SCHttpUnknownException {
+    return httpRequest(url, header, body, HTTPX_DELETE_STRING, outputFile);
+  }
+
+  /**
+   * Performs an HTTP OPTIONS request to the specified URL using the specified request header and
+   * body.
+   *
+   * @param url URL to make request
+   * @param header request header
+   * @param body request body
```

Comment:
options request does not have a body. I think it makes the most sense to remove parameter

Reply:
Removed (replaced with "")